collect_app/src/androidTest/java/org/odk/collect/android/feature/projects/GoogleDriveDeprecationBannerTest.kt

```diff
@@ -0,0 +1,125 @@
+import org.odk.collect.androidtest.RecordedIntentsRule
+
+class GoogleDriveDeprecationBannerTest {
+    val rule = CollectTestRule()
+    private val testDependencies = TestDependencies()
+
+    @get:Rule
+    val chain: RuleChain = TestRuleChain.chain(testDependencies)
+        .around(RecordedIntentsRule())
+        .around(rule)
+
+    @Test
+    fun bannerIsNotVisibleInNonGoogleDriveProjects() {
+        rule
+            .startAtMainMenu()
+            .assertTextNotDisplayed(R.string.google_drive_deprecation_message)
+    }
+
+    @Test
+    fun bannerIsVisibleInGoogleDriveProjects() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .assertText(R.string.google_drive_deprecation_message)
+    }
+
+    @Test
+    fun forumThreadIsOpenedAfterClickingLearnMore() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .clickOnString(R.string.learn_more_button_text)
+
+        intended(
+            allOf(
+                hasComponent(WebViewActivity::class.java.name),
+                hasExtra("url", "https://forum.getodk.org/t/40097")
+            )
+        )
+    }
+
+    @Test
+    fun dismissButtonIsVisibleOnlyAfterClickingLearnMore() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .assertTextNotDisplayed(R.string.dismiss_button_text)
+            .clickOnString(R.string.learn_more_button_text)
+            .pressBack(MainMenuPage())
+            .assertText(R.string.dismiss_button_text)
+    }
+
+    @Test
+    fun afterClickingDismissTheBannerDisappears() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .clickOnString(R.string.learn_more_button_text)
+            .pressBack(MainMenuPage())
+            .clickOnString(R.string.dismiss_button_text)
+            .assertTextNotDisplayed(R.string.google_drive_deprecation_message)
+            .rotateToLandscape(MainMenuPage())
```

Comment:
Is the rotation added to attempt to make sure that banner stays dismissed between different instances of `MainMenuActivity`? I'm finding that I can comment out the `unprotectedSettings.save(ProjectKeys.GOOGLE_DRIVE_DEPRECATION_BANNER_DISMISSED, true);` line and this test will still pass. Maybe the assertions don't wait long enough?

Rotation is definitely a little flakey: I'm wondering if a stabler approach would be to write a test for `MainMenuActivity` that checks it doesn't show the banner if `ProjectKeys.GOOGLE_DRIVE_DEPRECATION_BANNER_DISMISSED` is true and then another test to check that clicking "Dismiss" sets it to true. Using that approach, you might be able to avoid any Espresso tests, but I'll leave how you slice that up to you.